File: src/ProjectR/content/scripts/spells/Ram.lua
function GetTargetType()
    return Single
end

function GetName()
    return "Ram"
end

function GetDescription()
    return "Ram your target causing damage proportional to your DEF."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 0
end

function GetDelay()
  return .5
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
    local damage = aDEF - dDEF
    d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/Ram.cs
﻿using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class Ram : SpellScriptBase
    {
        public override string Name { get { return "Ram"; } }
        public override string Description { get { return "Ram your target causing damage proportional to your DEF."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 0; } }
        public override double Delay { get { return .5; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = aDEF - dDEF;
            DealDamage(damage);
        }
    }
}